### PR TITLE
[Work In Progress] - Add sound player - closes #1, closes #2, closes #3, closes #6

### LINKED ISSUE #1: Research audio player api

### LINKED ISSUE #2: apply for a soundcloud api key

### LINKED ISSUE #3: spend an hour looking at other music provider apis?

### LINKED ISSUE #6: Add ability to add/store mp3
Potentially add other audio file formats during this step as well

### PR DESCRIPTION
**Pull request by Jeff-Duke**
Purpose

Create a player component in Vue.  Configure it to play an MP3 from the local file system.

Approach
After researching several different audio libraries on NPM we opted to just use the HTML5 audio tag that is built in with chrome.  

Learning
Consulted electron docs, chrome docs, dataurl docs.  

Open Questions and Pre-Merge TODOs
Our dataurl function creates the proper buffer but is giving an error when trying to create a dataurl to pass to the audio source.

### CODE REVIEW

**Comment by Jeff-Duke** [comment]
@stevekinney we're to the point now where the src is being replaced with the dataurl but the dataurl has some extra info in the beginning 
screen shot 2016-12-08 at 21 20 01

Seems hacky to try to slice those chars off the front.  Are we doing something wrong in converting that dataurl?  I thought it was working when we ran it in Node...

**Comment by stevekinney** [comment]
Hot take before I look deeper at the code: Pretty sure that `undefined` is supposed to be file type (in this case: `"audio/mp3"`).

File: app/electron.js [hunk 1]
```diff
@@ -1,12 +1,16 @@
...
+const BrowserWindow = electron.BrowserWindow;
 
-let mainWindow
-let config = {}
+const gimmeSong = global.gimmeSong = require('./gimme-song-url');
+
+let mainWindow;
+let config = {};
+
+app.commandLine.appendSwitch("js-flags", "--allow-running-insecure-content");
```
Comment (stevekinney) [inline app/electron.js]: This should go.

File: app/gimme-song-url.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+const dataurl = require('dataurl');
+const fs = require('fs');
+
+const gimmeSong = (callback) => {
+  fs.readFile('/Users/Duke/turing/mod-4/projects/test-project/04-the_xx-islands.mp3',
+  (err, data) => {
+    if (err) { callback(err); }
+    const encoded = dataurl.convert({ data, mimeType: 'audio/mp3' });
```
Comment (stevekinney) [inline app/gimme-song-url.js]: I just checked the docs. It's `mimetype` with a lowercase "t", not `mimeType`.

Comment (bgdfry) [reply app/gimme-song-url.js]: I hate everything. That worked.

**Comment by bgdfry** [comment]
🙏🏼

**Pull request merged**